packages/ui/nodes/image/src/ImageElement/ImageElement.types.ts

```diff
@@ -21,33 +22,31 @@ export interface ImageElementStyles {
   handleRight: CSSProp;
 }
 
-export interface ImageElementProps<V extends Value>
-  extends StyledElementProps<V, TImageElement, ImageElementStyles> {
-  resizableProps?: ResizableProps;
+export interface ImageElementPropsCaption {
+  disabled?: boolean;
 
   /**
-   * Image alignment.
+   * Caption alignment.
    */
   align?: 'left' | 'center' | 'right';
 
-  caption?: {
-    disabled?: boolean;
+  /**
+   * Caption placeholder.
+   */
+  placeholder?: string;
 
-    /**
-     * Caption alignment.
-     */
-    align?: 'left' | 'center' | 'right';
+  /**
+   * Whether caption is read-only.
+   */
+  readOnly?: boolean;
+}
 
-    /**
-     * Caption placeholder.
-     */
-    placeholder?: string;
+export interface ImageElementProps<V extends Value>
```

Comment:
What are your thoughts on union types vs interfaces?

I know types can not be extended, while interfaces can. I tend to think of encouraging composition over inheritance.